mono/utils/mono-time.c

```diff
@@ -15,16 +15,23 @@
 #include <utils/mono-time.h>
 
 
-#define MTICKS_PER_SEC 10000000
+#define MTICKS_PER_SEC (10 * 1000 * 1000)
+
+gint64
+mono_msec_ticks (void)
+{
+	return mono_100ns_ticks () / 10 / 1000;
+}
 
 #ifdef HOST_WIN32
 #include <windows.h>
+#include <winbase.h>
 
-guint32
-mono_msec_ticks (void)
+gint64
+mono_msec_boottime (void)
 {
 	/* GetTickCount () is reportedly monotonic */
-	return GetTickCount ();
+	return GetTickCount64 ();
```

Comment:
Looks like the mingw headers don't include GetTickCount64, so we probably need to define it ourselves.